core/braintree/src/braintree_address.erl

```diff
@@ -92,6 +92,7 @@ update(#bt_address{id=AddressId
 -spec delete(bt_address()) -> bt_address().
 -spec delete(ne_binary() | nonempty_string(), ne_binary() | nonempty_string()) ->  bt_address().
 
+delete(#bt_address{id='undefined'}) -> #bt_address{};
```

Comment:
@fenollp will do so i was thinking of adding the check to multiple apis and thought every api will have to look here, but i was wrong and now i think this is wrong :p will do as suggested